### PR TITLE
If plot ranges cannot be determined use mpl autoscale view

### PR DESCRIPTION
**Pull request by philippjfr**
This ensures that the xlim and ylim on a plot correspond to the actual data limits at all times, allowing calculations based on these values to be performed.

### CODE REVIEW

File: holoviews/plotting/mpl/element.py [hunk 1]
```diff
@@ -314,16 +315,21 @@ def _set_axis_limits(self, axis, view, subplots, ranges):
...
             if self.invert_yaxis or any(p.invert_yaxis for p in subplots):
                 t, b = b, t
             if b != t:
                 if valid_lim(b):
                     axis.set_ylim(bottom=b)
+                    yset = True
                 if valid_lim(t):
                     axis.set_ylim(top=t)
+                    yset = True
+        axis.autoscale_view(scalex=not xset, scaley=not yset)
```
Comment (jlstevens) [inline holoviews/plotting/mpl/element.py]: Seems like a reasonable change although why not invert the booleans and use the same names as matplotlib ``scalex`` and ``scaley``? That would seem clearer to me...

Comment (philippjfr) [inline holoviews/plotting/mpl/element.py]: Agreed.

**Comment by philippjfr** [comment]
Apparently some ``Spikes`` plots are affected by this, will have to investigate that before this can be merged.

**Comment by philippjfr** [comment]
Ready to merge.

**Comment by philippjfr** [comment]
Not sure why the reference data is complaining but this is still ready to merge.

**Comment by jlstevens** [comment]
With the pr build passing, I'll merge it now.

**Pull request merged**